query/src/main/java/org/infinispan/search/mapper/search/loading/context/impl/InfinispanLoadingContext.java

```diff
@@ -62,7 +50,13 @@ public InfinispanLoadingContext build() {
       }
    }
 
-   private class InfinispanSelectionLoadingStrategy<E> implements PojoSelectionLoadingStrategy<E> {
+   public class InfinispanSelectionLoadingBinder {
```

Comment:
This can't be a nested class, it must be static since it'll have to be instantiated from `InfinispanEntityTypeContributor`.

Within the strategy itself, you'll be able to access `InfinispanLoadingContext` when `createEntityLoader` gets called: that's the `pojoSelectionLoadingContext` argument, you just need to cast it. From there you can access the `entityLoader` and return it.

Yes it's a bit odd; usually the entityLoader would be instantiated by the strategy. But that's how it's done in Infinispan :shrug: